File: java/src/Polimorfismo/Cilindro.java
package Polimorfismo;

public class Cilindro extends Figuras {

    Cilindro(String nome, int dim1, int dim2) {
        this.nome = nome;
        this.dim1 = dim1;
        this.dim2 = dim2;

    }

    public double volume() {
        double volume;
        volume = Math.PI * Math.pow(dim1, 2) * dim2;
        return volume;
    }

    public double area() {
        double area;
        area = (2 * Math.PI * dim1) * (dim1 + dim2);
        return area;
    }
    void display(){
        System.out.println(nome + dim1 + dim2);
    }
}

File: cpp/Polimorfismo/Cilindro.cpp
#include <cmath>
#include <iostream>
#include "Figuras.h"

class Cilindro : public Figuras {
public:
    Cilindro(std::string nome, double dim1, double dim2) : Figuras(nome, dim1, dim2) {}

    double volume() override {
        return M_PI * std::pow(dim1, 2) * dim2;
    }

    double area() override {
        return (2 * M_PI * dim1) * (dim1 + dim2);
    }

    void display() override {
        std::cout << nome << ": Raio = " << dim1 << ", Altura = " << dim2 << std::endl;
    }

    void aumentar() override {
        dim1 *= aumento;
        dim2 *= aumento;
    }
};
